cogs/music.py

```diff
@@ -239,15 +241,17 @@ async def play(self, ctx: commands.Context, query: str=None):
 			await ctx.send(f"You have to provide a youtube url or query. Type `{BOT_PREFIX}play <url/query>.")
 			return
 
-
 		# To avoid clutter, we edit the user's message to suppress the embed
 		msg = ctx.message
 		await msg.edit(suppress=True)
 
-		# to use search instead of passing an url, prepend 'ytsearch:' to the search terms
-		# to search for more than one video, prepend 'ytsearchx:' with x being the number of videos
-		video = ytdl.extract_info(f"ytsearch:{query}", download=False)['entries'][0]
-		url = video['webpage_url']
+		# Checks if query is a valid url, if not we search youtube for the query
+		if validate_url(query) is False:
```

Comment:
Since `validate_url` returns a bool, the test should be `if not await validate_url(query)`.